Sources/Purchasing/IntroEligibilityCalculator.swift

```diff
@@ -36,33 +36,27 @@ class IntroEligibilityCalculator {
         }
         Logger.debug(Strings.customerInfo.checking_intro_eligibility_locally)
 
-        var result = candidateProductIdentifiers.reduce(into: [:]) { resultDict, productId in
-            resultDict[productId] = IntroEligibilityStatus.unknown
-        }
+        var result = candidateProductIdentifiers.dictionaryWithValues { _ in IntroEligibilityStatus.unknown }
         do {
             let receipt = try self.receiptParser.parse(from: receiptData)
             Logger.debug(Strings.customerInfo.checking_intro_eligibility_locally_from_receipt(receipt))
 
-            let purchasedProductIdsWithIntroOffersOrFreeTrials =
-                receipt.purchasedIntroOfferOrFreeTrialProductIdentifiers()
-
-            let allProductIdentifiers =
-                candidateProductIdentifiers.union(purchasedProductIdsWithIntroOffersOrFreeTrials)
+            let activeSubscriptionProductIdentifiers = receipt.activeSubscriptionProductIdentifiers
+            let allProductIdentifiers = candidateProductIdentifiers.union(activeSubscriptionProductIdentifiers)
 
-            productsManager.products(withIdentifiers: allProductIdentifiers) {
+            self.productsManager.products(withIdentifiers: allProductIdentifiers) {
                 let allProducts = $0.value ?? []
 
-                let purchasedProductsWithIntroOffersOrFreeTrials = allProducts.filter {
-                    purchasedProductIdsWithIntroOffersOrFreeTrials.contains($0.productIdentifier)
-                }
                 let candidateProducts = allProducts.filter {
                     candidateProductIdentifiers.contains($0.productIdentifier)
                 }
+                let activeSubscriptionProducts = allProducts.filter {
+                    activeSubscriptionProductIdentifiers.contains($0.productIdentifier)
+                }
 
-                let eligibility = self.checkEligibility(
-                    candidateProducts: candidateProducts,
-                    purchasedProductsWithIntroOffers: purchasedProductsWithIntroOffersOrFreeTrials)
-                result.merge(eligibility) { (_, new) in new }
+                let eligibility = self.checkEligibility(candidateProducts: candidateProducts,
+                                                        activeSubscriptionProducts: activeSubscriptionProducts)
+                result.merge(eligibility, strategy: .overwriteValue)
```

Comment:
Simplified this using the `MergeStrategy` (see #2190)

Reply:
how does this differ from `result += eligibility`?

Reply:
Oh it's the same, I'll change it, thanks!
I improved this from the previous implementation after bae4a004 but didn't realize it can be even simpler now.